From: Guilhem Moulin <guilhem@debian.org>
Date: Sat, 16 Jul 2022 12:52:47 +0200
Subject: Pass $(LDFLAGS) when building fake_token_path.so

The Debian tooling sets ‘LDFLAGS = -Wl,-z,relro -Wl,-z,now’ and blhc(1p)
complains when anything is built without hardened compiler/linker flags.

Granted this is a non-issue here since fake_token_path.so isn't included
in any binary package, but muting the false positive is arguably no
better fix than honoring $(LDFLAGS) during the build.

Forwarded: https://gitlab.com/cryptsetup/cryptsetup/-/merge_requests/390
---
 tests/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index f49361c..edd060c 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -42,7 +42,7 @@ endif
 ssh-plugin-test: fake_token_path.so
 
 fake_token_path.so:
-	$(CC) $(CFLAGS) -I $(top_srcdir)/lib -fPIC -shared \
+	$(CC) $(CFLAGS) $(LDFLAGS) -I $(top_srcdir)/lib -fPIC -shared \
 	-Wl,--version-script=$(top_srcdir)/lib/libcryptsetup.sym \
 	-o fake_token_path.so  $(top_srcdir)/tests/fake_token_path.c \
 	-DBUILD_DIR=\"$(abs_top_srcdir)/.libs/\"
